doc/nrf/app_approtect.rst

```diff
@@ -0,0 +1,110 @@
+.. _app_approtect:
+
+Enabling access port protection mechanism
+#########################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+.. app_approtect_info_start
+
+Several Nordic Semiconductor SoCs supported in the |NCS| offer an implementation of the access port protection mechanism (APPROTECT).
+When enabled, this mechanism blocks the debugger from read and write access to all CPU registers and memory-mapped addresses.
+Accessing these registers and addresses again requires disabling the mechanism and erasing the flash.
+In factory state, all devices come with the access port protection enabled.
+
+.. app_approtect_info_end
+
+The following table lists SoCs with information about the APPROTECT mechanism they support.
+
+.. list-table:: SoC APPROTECT matrix
+   :header-rows: 1
+
+   * - SoC
+     - Related hardware revision that supports APPROTECT
```

Comment:
Regular AP-Protect is dtillHardened / improved AP-Protect is separate from AP-Protect as a whole